client/components/tinymce/plugins/wpcom-track-paste/plugin.js

```diff
@@ -46,39 +42,38 @@ function pasteTrack( editor ) {
 		return SOURCE_UNKNOWN;
 	}
 
-	function flattenArray( orgTypes ) {
-		// As per https://html.spec.whatwg.org/multipage/interaction.html#datatransfer
-		// types should be an array containing the available formats that were set on copy.
-		// Some browsers, as Firefox, return a DOMStringList instead, see
-		// https://developer.mozilla.org/en-US/docs/Web/API/DOMStringList
-		let types = [];
-		if ( orgTypes.contains ) {
-			types = Array.from( orgTypes );
-		} else {
-			types = orgTypes.slice();
-		}
-		return types.reduce( ( a, b ) => {
-			return a + ', ' + b;
-		} );
+	function toString( types ) {
+		// types should be an array. Some browsers, as Firefox, return a DOMStringList instead.
+		// We need coverage for thoses cases, hence the Array.from conversion.
+		//
+		// https://html.spec.whatwg.org/multipage/interaction.html#datatransfer
+		// DOMStringListhttps://developer.mozilla.org/en-US/docs/Web/API/DOMStringList
+		return Array.from( types ).join( ', ' );
 	}
 
-	function onPasteEditor( event ) {
-		recordEvent( flattenArray( event.clipboardData.types ), 'tinymce' );
+	function onPasteFromTinyMCEEditor( event ) {
+		recordEvent( 'tinymce-editor', toString( event.clipboardData.types ) );
 	}
 
-	function onPasteTextArea( event ) {
-		recordEvent( flattenArray( event.clipboardData.types ), 'html' );
+	function onPasteFromHTMLEditor( event ) {
+		recordEvent( 'html-editor', toString( event.clipboardData.types ) );
 	}
 
-	// Bind paste event listeners to both Visual and HTML editors
-	editor.on( 'paste', onPasteEditor );
+	debug( 'bind listeners for paste event' );
```

Comment:
we probably want to do a full review over which debug statements are valuable to us before we merge.

my guess is that many of these are more useful for the development period building this feature than they will be to leave in there after development has finished.